src/main.ts

```diff
@@ -193,10 +193,13 @@ app.whenReady().then(
 // Quit when all windows are closed, except on macOS. There, it's common
 // for applications and their menu bar to stay active until the user quits
 // explicitly with Cmd + Q.
-app.on('window-all-closed', () => {
+app.on('window-all-closed', async () => {
   if (process.platform !== 'darwin') {
     app.quit()
+    return
   }
+
+  await closeWatcher()
```

Comment:
The issue is related to chokidar still watching the main window that no longer exists:

https://github.com/grafana/k6-studio/blob/b818b0c6ef38469b77b7e61ce1d6f7d8697d8c47/src/main/watcher.ts#L40